src/webapp/components/import-export/ImportTable.tsx

```diff
@@ -0,0 +1,607 @@
+        };
+        loading.show(true);
+
+        const fetchData = async () => {
+            setIsLoading(true);
+            if (!allUsers) {
+                return;
+            }
+            const existingUsersMapped = _.keyBy(allUsers, getUsername) as Record<string, User>;
+            setExistingUsers(existingUsersMapped as unknown as SetStateAction<Record<string, User>>);
+            setExistingUsersNames(allUsers.map((user: User) => getUsername(user)));
+            setIsLoading(false);
+            loading.reset();
+        };
+
+        fetchData();
+    }, [allUsers, loading]);
+
+    const existingUserInTable = useCallback(
+        (newUsers: User[]) => {
+            if (!existingUsersNames) {
+                return false;
+            }
+            return _(newUsers).some(user => existingUsersNames.includes(user.username));
+        },
+        [existingUsersNames]
+    );
+
+    const closeInfoDialog = () => {
+        setInfoDialog(undefined);
+    };
+
+    const toggleAllowOverwrite = useCallback(
+        (_event, newValue: boolean) => {
+            setAreUsersValid(newValue || !errorsCount);
+            setAllowOverwrite(newValue);
+        },
+        [errorsCount]
+    );
+
+    const renderDialogTitle = () => {
+        const errorText =
+            errorsCount === 0
+                ? null
+                : i18n.t("{{n}} invalid users found, check in-line errors in table", { n: errorsCount });
+        const maxWarnings = 10;
+        const hiddenWarnings = Math.max(warnings.length - maxWarnings, 0);
+
+        const warningText =
+            warnings.length === 0
+                ? null
+                : _([
+                      i18n.t("{{n}} warning(s) while importing file", { n: warnings.length }) + ":",
+                      ..._(warnings)
+                          .take(maxWarnings)
+                          .map((line, idx) => `${idx + 1}. ${line}`)
+                          .value(),
+                      hiddenWarnings > 0 ? i18n.t("and_n_more_warnings", { n: hiddenWarnings }) : null,
+                  ])
+                      .compact()
+                      .join("\n");
+
+        return (
+            <React.Fragment>
+                <StyledDialogTitle>{title}</StyledDialogTitle>
+                {errorText && (
+                    <DialogTooltip title={errorText}>
+                        <FontIcon className="material-icons">error</FontIcon>
+                    </DialogTooltip>
+                )}
+                {warningText && (
+                    <DialogTooltip title={warningText}>
+                        <FontIcon className="material-icons">warning</FontIcon>
+                    </DialogTooltip>
+                )}
+            </React.Fragment>
+        );
+    };
+
+    const onSubmit = useCallback(
+        async ({ users }: { users: User[] }) => {
+            loading.show(true, i18n.t("Importing users"));
+            const { error } = await onSave(users);
```

Comment:
We can use the composition root here, right? Instead passing the method as a prop.

Reply:
I can call it here but I've seen that ImportTable was also used for in `ReplicateUserFromTable.component`, that's why title and other props are configurable. Should I call composition root here and make it more generic when we migrate `ReplicateUserFromTable` ?